demos/common/cpp/pipelines/src/async_pipeline.cpp

```diff
@@ -67,15 +67,17 @@ int64_t AsyncPipeline::submitData(const InputData& inputData, const std::shared_
     if (!request)
         return -1;
 
+    auto startTime = std::chrono::steady_clock::now();
     auto internalModelData = model->preprocess(inputData, request);
+    preprocessMetrics.update(startTime);
 
     request->SetCompletionCallback(
-        [this, frameID, request, internalModelData, metaData] {
+        [this, frameID, request, internalModelData, metaData, startTime] {
             request->SetCompletionCallback([]{});
 
             {
                 std::lock_guard<std::mutex> lock(mtx);
-
+                this->inferenceMetrics.update(startTime);
```

Comment:
```suggestion
                inferenceMetrics.update(startTime);
```
I'd prefer using `this` everywhere, but ... :)